lib/src/tasks/coverage/api.dart

```diff
@@ -317,13 +407,79 @@ class CoverageTask extends Task {
     return coverage;
   }
 
+  List<int> observatoryPort;
+
   Future _run() async {
     if (_html && !(await platform_util.isExecutableInstalled('lcov'))) {
       _done.completeError(new MissingLcovException());
       return;
     }
 
-    await _collect();
+    TaskProcess serve;
+
+    if(functional_tests.isNotEmpty){
+      //    String executable = 'selenium-server';
+//    int port = await getOpenPort();
+      List args = ["-port"
+//    ,"${port}"
+      ];
+      args = [];
+      observatoryPort = new List<int>();
+      RegExp _serving = new RegExp("Serving .* on http:\/\/localhost:8080");
+      serve = new TaskProcess('pub', ['serve']);
+      Completer serverRunning = new Completer();
+      serve.stderr.listen((line) async {
+        print(line);
+        if(line.contains("Error: Address already in use")){
+          await serve.kill();
+          await _seleniumServerProcess.kill();
+          throw new PortBoundException("pub serve failed to start.  Check if port 8080 is already bound");
+        }
+      });
+      serve.stdout.listen((line){
+        print(line);
+        if(_serving.hasMatch(line)){
+          serverRunning.complete();
+        }
+      });
+
+      _seleniumServerProcess = new TaskProcess('selenium-server', args);
```

Comment:
If possible, I'd like to avoid adding this dependency to this task. This forces the coverage task to be aware of functional tests and their requirement on a selenium server. While that is the case for our usage, it's possible that other consumers may run functional tests in a different way and without the need for a selenium server. Could we instead make this the responsibility of the project via the `before`/`after` configuration for the coverage task? For example..

```dart
main() async {
  config.coverage
    ..before = [_startSeleniumServer]
    ..after = [_stopSeleniumServer];
}

_startSeleniumServer() async { ... }
_stopSeleniumServer() async { ... }
```